### ISSUE TITLE
we should not filter students predictions

### ISSUE THREAD

**Issue by jvkubm** [constantinpape/torch-em#204]
https://github.com/constantinpape/torch-em/blob/a969237cc807cfa575545c379fedd628ce4079a6/torch_em/self_training/loss.py#L29
filtering students prediction caused my model to predict far too many voxels/lesions than it should. I think it should not be done here. Same in line 57

**Comment by anwai98** [comment]
Hmm okay. I think you can update this by not passing the label filters in the [`MeanTeacherTrainer`](https://github.com/constantinpape/torch-em/blob/a969237cc807cfa575545c379fedd628ce4079a6/torch_em/self_training/mean_teacher.py#L227) (and similar in the lines below for the other training and validation functions)

or, an easier option would be to update the loss (and metric) functions to not consider the label filter at all, regardless of it's passed to the `__call__` method or not.

**Comment by jvkubm** [comment]
yes I simply don't multiply prediction with label_filter, but wanted to let you know, coz it seems to me it should be formulated like this:
`loss = self.loss(prediction, labels * label_filter) `
 or am I missing sth?

**Comment by anwai98** [comment]
You don't need to mask the labels either. I think this should suffice:
```
loss = self.loss(predictions, labels)
```
